test cases/common/264 implementation/meson.build

```diff
@@ -0,0 +1,7 @@
+project('meson.implementation()',
+  default_options: {
+    'backend': 'null',
+  },
+)
+
+assert(meson.implementation() == 'meson')
```

Comment:
@annacrombie already does, and I'll eventually also, import all of the Meson tests. To make this easier, can we use an option for this that defaults to `'meson'`, and then we can have a simpler override, like adding a test.json or something?

Reply:
```meson
option('implementation', type: 'string', default: '')
```

This is along the lines of what you want to accompany the test with?

Reply:
See latest push